sonar-java-plugin/src/main/java/org/sonar/plugins/java/JavaPlugin.java

```diff
@@ -82,8 +87,19 @@ public void define(Context context) {
       JavaSquidSensor.class,
       PostAnalysisIssueFilter.class,
       XmlFileSensor.class);
-    context.addExtensions(builder.build());
 
+    if (isSupported(context.getRuntime())) {
+      builder.add(AnalysisWarningsWrapper.class);
+    }
+
+    context.addExtensions(builder.build());
   }
 
+  /**
+   * Drop this and related when the minimum supported version of SonarJava reaches 7.4.
+   */
+  private static boolean isSupported(SonarRuntime runtime) {
```

Comment:
the name of the method is a bit too generic, maybe this should be isAnalysisWarningsSupported ?